packages/cli/src/rollup-plugins/typescript-declarations/index.ts

```diff
@@ -22,12 +22,14 @@ export default function typescriptDeclarations(pkg: Package): Plugin {
       let deps = creator.getDeps(pkg.entrypoints.map((x) => x.source));
       await Promise.all(
         [...deps].map(async (dep) => {
-          let { name, content } = await creator.getDeclarationFile(dep);
-          srcFilenameToDtsFilenameMap.set(normalizePath(dep), name);
-          this.emitFile({
-            type: "asset",
-            fileName: path.relative(opts.dir!, name),
-            source: content,
+          let files = await creator.getDeclarationFiles(dep);
+          files.forEach(({ name, content }) => {
+            srcFilenameToDtsFilenameMap.set(normalizePath(dep), name);
```

Comment:
this has the potential of overriding the `.d.ts` value with `.d.ts.map`. I have no idea how this map is used so not sure how this can affect the program but looks like maybe something that should be addressed

Reply:
This should only set it for the `.d.ts` one, it used used futher down in this file to determine what the declaration re-export should point to.

Reply:
(i suspect it happens to work right now because the dts map comes first in the array but this should still be fixed)

Reply:
> (i suspect it happens to work right now because the dts map comes first in the array but this should still be fixed)

Yep, that was my suspicion as well.